src/esm_parser/esm_parser.py

```diff
@@ -1527,6 +1527,85 @@ def resolve_choose(model_with_choose, choose_key, config):
         logging.debug("key=%s", key)
 
 
+def resolve_early_var(
+    var, config, current_model=None, user_config={}, model_config={}, setup_config={}
+):
+
+    #import ipdb
+    if not current_model:
```

Comment:
Not sure about this, but maybe you need something like this instead:
```suggestion
    if current_model is None:
```

As currently, it is you check if it is not a "False"-y thing.